src/routes/api/bookmarks/bookmarkController.js

```diff
@@ -0,0 +1,117 @@
+import db from '../../../models';
+
+const { bookmark } = db;
+/**
+ *contains static methods for adding, removing
+ *and getting bookmarks
+ * @class Bookmarks
+ */
+class Bookmark {
+  /**
+   *method to handle adding bookmarks
+   *to articles
+   * @static
+   * @param {Request} req object
+   * @param {Response} res object
+   * @returns {void} void
+   * @memberof Bookmark
+   */
+  static async addBookmark(req, res) {
+    const { article } = req.body;
+    try {
+      const bookmarks = await bookmark.create({
+        user: req.user.userid,
+        article
+      });
+      return res.status(201).json({
+        status: 201,
+        bookmarks
+
+      });
+    } catch (err) {
+      // Errorcode for Sequelize fkey constraint = 23503
+      if (err.parent.code === '23503') {
+        return res.status(404).json({
+          status: 404,
+          error: 'Article not found'
+        });
+      }
+      // Error code for Sequelize unique constraint = 23505
+      if (err.parent.code === '23505') {
+        return res.status(409).json({
+          status: 409,
+          message: 'You have already bookmarked this article'
+        });
+      }
+      return res.status(500).json({
+        status: 500,
+        error: 'Internal server error'
+      });
+    }
+  }
+
+  /**
+ *method to handle removing bookmarks from
+ *articles
+ * @static
+ * @param {Request} req object
+ * @param {Response} res object
+ * @returns {void} void
+ * @memberof Bookmark
+ */
+  static async removeBookmark(req, res) {
+    const { article } = req.body;
+    const user = req.user.userid;
+    try {
+      await bookmark.destroy({
+        where: { user, article }
+      });
+      return res.status(200).json({
+        status: 200,
+        message: 'Bookmark removed'
+      });
+    } catch (err) {
+      return res.status(500).json({
+        status: 500,
```

Comment:
The error handling here is ambiguous. KIndly review this code and provide a more robust and descriptive error handling functionality.

Reply:
Fixed.